test/requests/formatting_test.rb

```diff
@@ -167,4 +193,11 @@ def clear_syntax_tree_runner_singleton_instance
 
     T.unsafe(Singleton).__init__(RubyLsp::Requests::Support::SyntaxTreeFormattingRunner)
   end
+
+  def stub_syntax_tree(present:)
```

Comment:
Rather than trying to manipulate the loaded gems to mirror Bundler's behaviour, this seemed more straightforward.